feat(BE): implement user update and migration setup (#23)
XtremeProgramming/typewrite: README.md

To explore the API documentation automatically generated by the API, go to `http://localhost:8080/docs` in your browser.

There, you’ll find a complete list of all available API endpoints, along with their request/response formats and the ability to test them directly from the browser.

### Migrations

To ensure your local database schema is up to date, you need to run the migration script provided in the project.
Follow the steps below to set the necessary permissions and execute the migration.

> **Important:** Before running the migration commands, make sure both the **backend** and **db** are up and running.

Reviewer: ```suggestion
**Important:** Before running the migration commands, make sure both the **api** and **db** are up and running.
```

This ensures that Alembic can connect to the database and apply the necessary changes without errors.

1. Gives execution permission to the migration script located at ./bin/migrate.sh.
   This command is usually run once after cloning the repository or creating the script, to ensure it can be executed as a program.

```bash
chmod +x ./bin/migrate.sh
```

2. Executes the migration script.

>    This script typically contains commands to generate and/or apply Alembic migrations (e.g., alembic upgrade head), automating the database update process.

Reviewer: ```suggestion
   This script applies the migrations created by alembic.
```
